WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -295,16 +295,21 @@ class PostCoordinator: NSObject {
             return
         }
 
+        let mediaLink = media.link
         let mediaUploadID = media.uploadID
         let gutenbergMediaUploadID = media.gutenbergUploadID
         if media.remoteStatus == .failed {
             return
         }
         if media.mediaType == .image {
-            let imgPostUploadProcessor = ImgUploadProcessor(mediaUploadID: mediaUploadID, remoteURLString: remoteURLStr, width: media.width?.intValue, height: media.height?.intValue)
-            postContent = imgPostUploadProcessor.process(postContent)
             let gutenbergImgPostUploadProcessor = GutenbergImgUploadProcessor(mediaUploadID: gutenbergMediaUploadID, serverMediaID: mediaID, remoteURLString: remoteURLStr)
             postContent = gutenbergImgPostUploadProcessor.process(postContent)
+
+            let gutenbergGalleryPostUploadProcessor = GutenbergGalleryUploadProcessor(mediaUploadID: gutenbergMediaUploadID, serverMediaID: mediaID, remoteURLString: remoteURLStr, mediaLink: mediaLink)
+            postContent = gutenbergGalleryPostUploadProcessor.process(postContent)
+
+            let imgPostUploadProcessor = ImgUploadProcessor(mediaUploadID: mediaUploadID, remoteURLString: remoteURLStr, width: media.width?.intValue, height: media.height?.intValue)
+            postContent = imgPostUploadProcessor.process(postContent)
```

Comment:
the ImgUploadProcessor (that should be used only on Aztec posts) needs to be last or it will change just the image information. I can add a check to see if the post is Gutenberg or not and split the processors for Gutenberg and Aztec.

Reply:
Hmm if we can make this less fragile that’d be great. I could easily add a new processor at the end without knowing ImgUploadProcessor needs to be last.

Reply:
I sorted this issue by refactoring the code to have two lists of processors one for GB blocks and another for Aztec and making sure that they are processed in the correct order:

I considered to add a check for GB post content, but the current way of doing this is so fragile, it's just a check for ```<!-- wp:``` that I think we better nor rely on that.

Commit: https://github.com/wordpress-mobile/WordPress-iOS/pull/13032/commits/7d9dafc56bc285fbd69cef3e678c33c58d1d54f1

Reply:
This should be sufficient for now, thanks!